File: Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/model/SystemModel.java
package org.cqframework.cql.cql2elm.model;

import org.hl7.cql.model.DataType;
import org.hl7.cql.model.SimpleType;
import org.hl7.elm_modelinfo.r1.ModelInfo;

public class SystemModel extends Model {
    public SystemModel(ModelInfo modelInfo) throws ClassNotFoundException {
        super(modelInfo, null);
    }

    public DataType getAny() {
        return this.resolveTypeName("Any");
    }

    public DataType getBoolean() {
        return this.resolveTypeName("Boolean");
    }

    public DataType getInteger() {
        return this.resolveTypeName("Integer");
    }

    public DataType getLong() {
        return this.resolveTypeName("Long");
    }

    public DataType getDecimal() {
        return this.resolveTypeName("Decimal");
    }

    public DataType getString() {
        return this.resolveTypeName("String");
    }

    public DataType getDateTime() {
        return this.resolveTypeName("DateTime");
    }

    public DataType getDate() {
        return this.resolveTypeName("Date");
    }

    public DataType getTime() {
        return this.resolveTypeName("Time");
    }

    public DataType getQuantity() {
        return this.resolveTypeName("Quantity");
    }

    public DataType getRatio() {
        return this.resolveTypeName("Ratio");
    }

    public DataType getCode() {
        return this.resolveTypeName("Code");
    }

    public DataType getConcept() {
        return this.resolveTypeName("Concept");
    }

    public DataType getVocabulary() {
        return this.resolveTypeName("Vocabulary");
    }

    public DataType getCodeSystem() {
        return this.resolveTypeName("CodeSystem");
    }

    public DataType getValueSet() {
        return this.resolveTypeName("ValueSet");
    }

    public DataType getVoid() {
        return new SimpleType("Void");
    }
}

File: Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/model/SystemModel.kt
package org.cqframework.cql.cql2elm.model

import org.hl7.cql.model.DataType
import org.hl7.cql.model.SimpleType
import org.hl7.elm_modelinfo.r1.ModelInfo

class SystemModel(modelInfo: ModelInfo) : Model(modelInfo, null) {
    val any: DataType
        get() = this.resolveTypeName("Any")!!

    val boolean: DataType
        get() = this.resolveTypeName("Boolean")!!

    val integer: DataType
        get() = this.resolveTypeName("Integer")!!

    val long: DataType
        get() = this.resolveTypeName("Long")!!

    val decimal: DataType
        get() = this.resolveTypeName("Decimal")!!

    val string: DataType
        get() = this.resolveTypeName("String")!!

    val dateTime: DataType
        get() = this.resolveTypeName("DateTime")!!

    val date: DataType
        get() = this.resolveTypeName("Date")!!

    val time: DataType
        get() = this.resolveTypeName("Time")!!

    val quantity: DataType
        get() = this.resolveTypeName("Quantity")!!

    val ratio: DataType
        get() = this.resolveTypeName("Ratio")!!

    val code: DataType
        get() = this.resolveTypeName("Code")!!

    val concept: DataType
        get() = this.resolveTypeName("Concept")!!

    val vocabulary: DataType
        get() = this.resolveTypeName("Vocabulary")!!

    val codeSystem: DataType
        get() = this.resolveTypeName("CodeSystem")!!

    val valueSet: DataType
        get() = this.resolveTypeName("ValueSet")!!

    val void: DataType
        get() = SimpleType("Void")
}
